src-tauri/src/gui/mod.rs

```diff
@@ -669,15 +668,22 @@ fn spawn_progress_monitor(
     thread::spawn(move || {
         let progress = ProgressBar::new(app_handle.clone(), &format!("Installing IDF {}", version));
 
+        let mut download_finished = false;
+
         while let Ok(message) = rx.recv() {
             match message {
                 ProgressMessage::Finish => {
                     progress.update(100, None);
                     progress.finish();
-                    break;
+                    download_finished = true;
                 }
                 ProgressMessage::Update(value) => {
+                  if download_finished {
+                    progress.update(value, Some(&format!("Downloading submodules...")));
+
```

Comment:
Can we show the user which submodule is being installed to provide a better visual experience, since this process usually takes time?